### ISSUE TITLE
[New Sniff] Check for removal of admin pages

### ISSUE THREAD

**Issue by jrfnl** [WPTRT/WordPress-Coding-Standards#17]
#### Rule:
Removal of WP admin pages is not allowed.

The exception would be a child theme removing a theme admin submenu page added by the parent theme.

Ref: https://github.com/WPTRT/WordPress-Coding-Standards/issues/12#issuecomment-231783316

#### Theme check file covering this rule:
n/a

##### Notes for implementation:
* Check for `remove_menu_page()` and forbid this.
* Check for `remove_submenu_page()` and only forbid this if one of a known list of WP Core admin pages is being removed

#### To do:
- [ ] Add the rule in the Theme Review handbook to the Requirements page.
- [ ] Create unit tests
- [ ] Create new sniff

**Comment by jrfnl** [comment]
Discussed in the meeting of May 31st, let's add this.

**Comment by jrfnl** [comment]
To create this sniff, I need some more information about the WP Core admin pages which are not supposed to be removed.

The question is basically: which submenu items are allowed to be removed ? Answers I can think of:
- Only admin pages within the `Themes` submenu and of those, not the known core ones (themes, customizer and its variants, widgets, menus, theme-editor).
- Admin pages within the `Themes` submenu + the `Options` submenu, again with the exception of the known core ones
- Any admin page, as long as its not a known core one

**Comment by justintadlock** [comment]
Essentially, the only admin pages allowed to be removed would be custom Appearance submenu items created by a parent theme and removed by a child theme.  I can't think of any others we allow to be removed.  I'm not even sure that there's a child theme that does this in the directory.  The situation would be rare.

**Comment by grappler** [comment]
I would go with
> Only admin pages within the Themes submenu and of those, not the known core ones (themes, customizer and its variants, widgets, menus, theme-editor).

I agree with Justin.

**Comment by jrfnl** [comment]
Thanks for the feedback. I have the sniff nearly ready already, but won't be able to get back to it until the weekend.

**Comment by dingo-d** [comment]
@jrfnl is this a different sniff from the forbidden functions one?

**Comment by jrfnl** [comment]
Yes it is and a lot more complicated one. Themes can removed admin pages from, for instance, a parent theme, but they are not allowed to remove WP Core admin pages, so there is a lot more involved in that.

I worked on it last it's quite tricky. I'd need to have a good look again to see where I got stuck.

**Comment by dingo-d** [comment]
I've removed the input/decision needed, as we know what is and isn't allowed.
